Sources/TracingInstrumentation/Span.swift

```diff
@@ -41,6 +41,12 @@ public protocol Span {
     /// - Parameter event: The `SpanEvent` to add to this `Span`.
     mutating func addEvent(_ event: SpanEvent)
 
+    /// Record an error of the given type described by the the given message.
+    /// - Parameters:
+    ///   - type: The type of error, e.g. "SomeError.case"
+    ///   - message: A message describing the error, e.g. "Division by zero"
+    mutating func recordError(ofType type: String?, message: String?)
```

Comment:
seems a bit manual, not `recordError(_ error: Error)`?

we can `type(of: error)` and "\(error)" record it I think?

The combination of the two usually renders pretty well.

Reply:
Not sure what you mean by combining them? Use `String(reflecting:...)` for type and `"\(error)"` for message? Or combine both values to create the kind?

Reply:
Just in the sense that if looking at this event in the backend and I have the type and the string repr, I have the full picture -- i.e. that reporting those two should be enough. But seems we're not in the field of making this call anyway, as the actual reporter should :)

Reply:
Yep, we'd still kind off have to implement it for the test `OTelSpan`, even though it's not of interest to us. I created #112 to rename it.

Reply:
Yup; btw maybe good to rename that one to some ExampleSpan? All of our Span are pretty much otel based in a way 🤔